tiny_dnn/layers/max_unpooling_layer.h

```diff
@@ -62,74 +60,77 @@ class max_unpooling_layer : public feedforward_layer<Activation> {
                       serial_size_t in_channels,
                       serial_size_t unpooling_size,
                       serial_size_t stride)
-    : Base({vector_type::data}),
+    : layer({vector_type::data}, {vector_type::data}),
       unpool_size_(unpooling_size),
       stride_(stride),
       in_(in_width, in_height, in_channels),
       out_(unpool_out_dim(in_width, unpooling_size, stride),
            unpool_out_dim(in_height, unpooling_size, stride),
            in_channels) {
-    // set_worker_count(CNN_TASK_SIZE);
+    // init_workers(CNN_TASK_SIZE);
     init_connection();
   }
 
-  size_t fan_in_size() const override { return 1; }
+  serial_size_t fan_in_size() const override { return 1; }
 
-  size_t fan_out_size() const override { return in2out_[0].size(); }
+  serial_size_t fan_out_size() const override { return in2out_[0].size(); }
 
-  void forward_propagation(serial_size_t index,
-                           const std::vector<vec_t *> &in_data,
-                           std::vector<vec_t *> &out_data) override {
-    const vec_t &in = *in_data[0];
-    // vec_t&       out = *out_data[0];
-    vec_t &a = *out_data[1];
-    std::vector<serial_size_t> &max_idx =
-      max_unpooling_layer_worker_storage_[index].in2outmax_;
+  void forward_propagation(const std::vector<tensor_t *> &in_data,
+                           std::vector<tensor_t *> &out_data) override {
+    const tensor_t &in = *in_grad[0];
+    tensor_t &out      = *out_grad[0];
```

Comment:
In original implementation `in` local variable references `in_data` and in this PR, it references `in_grad`.
How kind that is for reviewers!